dotnet3.5/S3/ListObjectsExample/ListObjects/ListObjects.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+// snippet-start:[S3.dotnet35.ListObjects]
+
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using System;
+using System.Threading.Tasks;
+
+namespace ListObjects
+{
+    // The following example will list the objects in an Amazon Simple Storage
+    // Service (Amazon S3) bucket. It was created using AWS SDK for .NET 3.5
+    // and .NET 5.0.
+    public class ListObjects
+    {
+        // Specify your AWS Region (an example Region is shown).
+        private static readonly RegionEndpoint BUCKET_REGION = RegionEndpoint.USWest2;
+        private static IAmazonS3 _s3Client;
+
+        // Specify the name of the new bucket
+        private const string BUCKET_NAME = "doc-example-bucket";
+
+        static void Main()
+        {
+            _s3Client = new AmazonS3Client(BUCKET_REGION);
+
+            Console.WriteLine($"Listing the objects contained in {BUCKET_NAME}:\n");
+            ListingObjectsAsync(_s3Client, BUCKET_NAME).Wait();
+        }
+
+        static async Task ListingObjectsAsync(IAmazonS3 client, string bucketName)
```

Comment:
Will this also list objects in any folders (and their child folders, and their child folders, etc.) inside the bucket as well?

Why limit this to 10? Also, would it be obvious to users in the output that only up to 10 objects are returned? Why not use the continuation tokens and paging features in the AWS SDK for .NET to make this more robust?

Reply:
S3 doesn't have folders. When you see one in the console, that's a convenience feature of the console.

Reply:
I created a folder and when I run the program, it lists the bucket and the files within it, like this listing:

![bucket_with_folder](https://user-images.githubusercontent.com/73654695/100625482-99476700-32f2-11eb-8907-97343a32b105.png)